grpc/bcodes.go

```diff
@@ -62,10 +67,25 @@ func errorToCode(err error) codes.Code {
 	}
 }
 
-func wrapError(err error) error {
+// wrapError wraps the internal error types we use for transport across the gRPC
+// layer. core.XXXError and probs.ProblemDetails error types are encoded using the gRPC
+// error status code which has been deprecated (#2507). errors.BoulderError error types
+// are encoded using the grpc/metadata in the context.Context for the RPC which is
+// considered to be the 'proper' method of encoding custom error types (grpc/grpc#4543
+// and grpc/grpc-go#478)
+func wrapError(ctx context.Context, err error) error {
 	if err == nil {
 		return nil
 	}
+	if berr, ok := err.(*berrors.BoulderError); ok {
+		// If setting the metadata fails ignore the returned error and
```

Comment:
This comment is hard to understand. How about: "Ignoring the error return here is safe because if setting the metadata fails, we'll still return an error, but it will be interpreted on the other side as an InternalServerError instead of a more specific one."